src/main/java/testapp/core/User.java

```diff
@@ -1,30 +1,40 @@
 package testapp.core;
 
-public class User {
-    private int id;
+import com.google.appengine.api.datastore.Entity;
+
+import java.io.Serializable;
+
+public class User implements Serializable {
+    private long ucn;
     private String name;
```

Comment:
user by him self got ucn as an object, entity has his id doesnt metter if it is user or account, id of entity is something like hashcode thats my understanding of difference between user id and entity id